src/main/java/com/basho/riak/client/core/codec/TermToBinaryCodec.java

```diff
@@ -322,30 +324,37 @@ private static Cell parseCell(List<RiakTsPB.TsColumnDescription> columnDescripti
     {
         if (cell instanceof OtpErlangBinary)
         {
-            OtpErlangBinary v = (OtpErlangBinary) cell;
-            String s = new String(v.binaryValue(), StandardCharsets.UTF_8);
-            return new Cell(s);
+            final OtpErlangBinary v = (OtpErlangBinary) cell;
+            if (columnDescriptions.get(j).getType() == RiakTsPB.TsColumnType.VARCHAR)
```

Comment:
IMO better to throw exception in case when something is incompatible, rather then silently skip it:

```
switch(columnDescriptions.get(j).getType()){
  case RiakTsPB.TsColumnType.VARCHAR://
  case RiakTsPB.TsColumnType.BLOB: //
  default:
    throw new IllegalStateException(<proper explanation>)
}